lib/src/components/NxSmallThreatCounter/NxSmallThreatCounter.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import classnames from 'classnames';
+import { head, repeat, tail, toUpper } from 'ramda';
+
+import { CounterProps, Props, propTypes } from './types';
+export { Props } from './types';
+
+import './NxSmallThreatCounter.scss';
+import NxTooltip from '../NxTooltip/NxTooltip';
+
+function Counter({ category, count, maxDigits }: CounterProps) {
+  const categoryForDisplay = `${toUpper(head(category))}${tail(category)}`,
+      className = classnames(`nx-small-threat-counter nx-small-threat-counter--${category}`, {
+        'nx-small-threat-counter--zero': count === 0
+      }),
+
+      // If we want maxDigits = 3 that means we want room for three digits in addition to a plus sign so
+      // that we can show e.g. "999+"
+      overflowDisplay = repeat('9', maxDigits).concat('+').join(''),
+      maxValue = Math.pow(10, Math.floor(maxDigits)) - 1;
+
+  return (
+    <NxTooltip title={categoryForDisplay}>
+      <div className={className}>
+        {/* Ideally this would be implemented as an aria-label but that doesn't get read by ChromeVox */}
+        <span className="nx-small-threat-counter__category">{categoryForDisplay}</span>
+        <span className="nx-small-threat-counter__count">{count > maxValue ? overflowDisplay : count}</span>
+        <div className="nx-small-threat-counter__sizer">
+          {overflowDisplay}
+        </div>
+      </div>
+    </NxTooltip>
+  );
+}
+
+export default function NxSmallThreatCounter(props: Props) {
+  const {
+        criticalCount,
+        severeCount,
+        moderateCount,
+        lowCount,
+        noneCount,
+        maxDigits: maxDigitsProp,
+        className: classNameProp,
+        ...attrs
+      } = props,
+      className = classnames('nx-small-threat-counter-container', classNameProp),
+      maxDigits = maxDigitsProp || 3;
+
+  if (maxDigits < 1) {
+    throw new Error('maxDigits must be positive');
+  }
+
+  if (typeof criticalCount !== 'number' &&
+      typeof severeCount !== 'number' &&
+      typeof moderateCount !== 'number' &&
+      typeof lowCount !== 'number' &&
+      typeof noneCount !== 'number') {
+    console.warn('No counts have been provided and so nothing will be rendered.');
+    return null;
+  }
+
+  return (
+    <div className={className} { ...attrs }>
```

Comment:
Although this component is primarily used within tables, I think it would still benefit from accepting `refs`?

Reply:
> Although this component is primarily used within tables, I think it would still benefit from accepting refs?

The thing is the main point of accepting refs (though theoretically not the only one) is to support tooltips.  But this component as a whole should not have a tooltip since each of it's visible parts has a built-in tooltip